File: src/main/java/com/carservice/service/FileService.java
package com.carservice.service;

import com.carservice.model.Car;
import com.carservice.model.ServiceRecord;
import com.carservice.model.User;
import com.fasterxml.jackson.core.type.TypeReference;
import com.fasterxml.jackson.databind.ObjectMapper;
import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;
import org.springframework.stereotype.Service;

import java.io.File;
import java.io.IOException;
import java.util.ArrayList;
import java.util.List;

@Service
public class FileService {
    private final ObjectMapper objectMapper;
    private final String usersFilePath = "users.txt";
    private final String carsFilePath = "cars.txt";
    private final String serviceHistoryFilePath = "service_history.txt";

    public FileService() {
        this.objectMapper = new ObjectMapper();
        this.objectMapper.registerModule(new JavaTimeModule());
        
        // Create files if they don't exist
        createFileIfNotExists(usersFilePath);
        createFileIfNotExists(carsFilePath);
        createFileIfNotExists(serviceHistoryFilePath);
    }

    private void createFileIfNotExists(String filePath) {
        File file = new File(filePath);
        if (!file.exists()) {
            try {
                file.createNewFile();
                if (filePath.equals(usersFilePath)) {
                    objectMapper.writeValue(file, new ArrayList<User>());
                } else if (filePath.equals(carsFilePath)) {
                    objectMapper.writeValue(file, new ArrayList<Car>());
                } else if (filePath.equals(serviceHistoryFilePath)) {
                    objectMapper.writeValue(file, new ArrayList<ServiceRecord>());
                }
            } catch (IOException e) {
                e.printStackTrace();
            }
        }
    }

    // User Management
    public List<User> loadUsers() {
        try {
            return objectMapper.readValue(new File(usersFilePath), new TypeReference<List<User>>() {});
        } catch (IOException e) {
            e.printStackTrace();
            return new ArrayList<>();
        }
    }

    public void saveUsers(List<User> users) {
        try {
            objectMapper.writeValue(new File(usersFilePath), users);
        } catch (IOException e) {
            e.printStackTrace();
        }
    }

    // Car Management
    public List<Car> loadCars() {
        try {
            return objectMapper.readValue(new File(carsFilePath), new TypeReference<List<Car>>() {});
        } catch (IOException e) {
            e.printStackTrace();
            return new ArrayList<>();
        }
    }

    public void saveCars(List<Car> cars) {
        try {
            objectMapper.writeValue(new File(carsFilePath), cars);
        } catch (IOException e) {
            e.printStackTrace();
        }
    }

    // Service History
    public List<ServiceRecord> loadServiceHistory() {
        try {
            return objectMapper.readValue(new File(serviceHistoryFilePath), new TypeReference<List<ServiceRecord>>() {});
        } catch (IOException e) {
            e.printStackTrace();
            return new ArrayList<>();
        }
    }

    public void saveServiceHistory(List<ServiceRecord> records) {
        try {
            objectMapper.writeValue(new File(serviceHistoryFilePath), records);
        } catch (IOException e) {
            e.printStackTrace();
        }
    }
}

File: lib/file-service.ts
// This service simulates file I/O operations by using localStorage in the browser
// In a real application, this would interact with actual files on the server

class FileService {
  // User Management
  async loadUsers() {
    try {
      const usersData = localStorage.getItem("users.txt")
      return usersData ? JSON.parse(usersData) : []
    } catch (error) {
      console.error("Error loading users:", error)
      return []
    }
  }

  async saveUsers(users) {
    try {
      localStorage.setItem("users.txt", JSON.stringify(users))
      return true
    } catch (error) {
      console.error("Error saving users:", error)
      throw error
    }
  }

  // Car Management
  async loadCars() {
    try {
      const carsData = localStorage.getItem("cars.txt")
      return carsData ? JSON.parse(carsData) : []
    } catch (error) {
      console.error("Error loading cars:", error)
      return []
    }
  }

  async saveCars(cars) {
    try {
      localStorage.setItem("cars.txt", JSON.stringify(cars))
      return true
    } catch (error) {
      console.error("Error saving cars:", error)
      throw error
    }
  }

  // Service History
  async loadServiceHistory() {
    try {
      const historyData = localStorage.getItem("service_history.txt")
      return historyData ? JSON.parse(historyData) : []
    } catch (error) {
      console.error("Error loading service history:", error)
      return []
    }
  }

  async saveServiceHistory(records) {
    try {
      localStorage.setItem("service_history.txt", JSON.stringify(records))
      return true
    } catch (error) {
      console.error("Error saving service history:", error)
      throw error
    }
  }
}

export const fileService = new FileService()

